### PR TITLE
New animations

### PR DESCRIPTION
**Pull request by alexvins**
New features
- support margins in .json animation
- new creatures animation types: distinct 2-hex attack and cast

### CODE REVIEW

**Comment by edeksumo** [comment]
For me creatures still don't play cast animation. And as I remember in OH, when creature don't have casting animation it will play coresponding attack anim.

**Comment by edeksumo** [comment]
Now it's working. But now that I'm looking on this, some creatures shouldn't play this animation, like wyverns and it's looking odd when them strike twice. I thought that mayby it shoulld be ranged animation used than meelee but archangels don't have it and as they resurect they play meelee animation.

**Comment by edeksumo** [comment]
So I look't into archangel def and he has anim for casting so when they don't have anim for casting they use ranged attack animation.

**Comment by edeksumo** [comment]
Now it's working allmost like in OH, but ogre magi don't play ani when casting (have animation), and for example basilsk (?) have animation to cast after attack but don't work.  And alternative death ani is working but is VERY lagy like in slow motion.

**Comment by alexvins** [comment]
There is no generic way to determine should animation be played after attack or not, I just blocked it.

**Comment by edeksumo** [comment]
There is no way to check that creature have animation to casting spell and play it? And what about ogre magi?

**Comment by alexvins** [comment]
In VCMI any creature can have any ability, so I just blocked animation for after and before attack cast. I`ll check ogre magi in a few miniutes

**Comment by misiokles** [comment]
I think that not all animations are displayed in battle hero defs also. For example - shuttle part of such defs. When battle is on and player doesn't do any click for some time, hero's horse start to 'breath' and 'shuttle' legs.

**Comment by alexvins** [comment]
@misiokles you`re right, only standing and cast actually used.

File: client/battle/CBattleInterface.cpp [hunk 1]
```diff
@@ -1416,9 +1416,11 @@ CBattleInterface::PossibleActions CBattleInterface::getCasterAction(const CSpell
 void CBattleInterface::setHeroAnimation(ui8 side, int phase)
 {
 	if(side == BattleSide::ATTACKER)
-		attackingHero->setPhase(phase);
+		if(attackingHero)
+			attackingHero->setPhase(phase);
 	else
-		defendingHero->setPhase(phase);
+		if(defendingHero)
+			defendingHero->setPhase(phase);
```
Comment (ArseniyShestakov) [inline client/battle/CBattleInterface.cpp]: I think we decide explicitly that something like that would need braces.

Comment (alexvins) [reply client/battle/CBattleInterface.cpp]: fixed

**Pull request merged**